Kernel/FileSystem/Ext2FileSystem.cpp

```diff
@@ -1499,9 +1499,9 @@ ErrorOr<NonnullRefPtr<Inode>> Ext2FS::create_inode(Ext2FSInode& parent_inode, St
     return new_inode;
 }
 
-ErrorOr<void> Ext2FSInode::populate_lookup_cache() const
+ErrorOr<void> Ext2FSInode::populate_lookup_cache()
 {
-    MutexLocker locker(m_inode_lock);
+    MutexLocker locker(m_lookup_cache_lock);
```

Comment:
There's no need for this lock, all callers already hold the `m_inode_lock` exclusively, instead assert that `m_inode_lock` is held.

Reply:
This is not true, because `Ext2FSInode::lookup` will try to use this method and as far as I can tell doesn't hold that lock when calling this method. We can change that but really I don't see why we should change the order of taking locks in the lookup method.